src/test/java/com/amplitude/EventTest.java

```diff
@@ -0,0 +1,45 @@
+package com.amplitude;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.junit.jupiter.api.Test;
+
+public class EventTest {
+  @Test
+  public void testCreateEventWithNullUserAndDeviceThrowsException() {
+    assertThrows(
+        IllegalArgumentException.class,
+        () -> {
+          new Event("test event", null, null);
+        },
+        "Event must have one defined userId and/or deviceId");
+  }
+
+  @Test
+  public void testToJsonObject() {
+    Event event = new Event("test event", "test-user");
+    String longStr = "Long string to be truncated.";
+    for (int i = 0; i < 7; i++) {
+      longStr += longStr;
+    }
+    assertEquals(3584, longStr.length());
+    JSONObject eventProperties = new JSONObject();
+    eventProperties.put("event_type", "test event type");
+    JSONArray eventMsgArray = new JSONArray();
+    eventMsgArray.put(longStr);
+    eventMsgArray.put(longStr);
+    eventProperties.put("event_message", eventMsgArray);
+    event.eventProperties = eventProperties;
+
+    JSONObject truncatedEvent = event.toJsonObject();
```

Comment:
Do we need to cover all possible keys in Event Class, and check if we can get the correct value from `event.toJsonObject()`?

Reply:
I don't mind just skipping it. @qingzhuozhen if you really, really wanted to test every single param. Otherwise, I'm just going to assume the json library has its own tests.

Reply:
I guess there is no need to set or assert all possible keys here, I add some assertion on the ones we are having values. Later we can cover more if needed.

Reply:
Sounds good.